### PR TITLE
[WIP] Migrated to PHP 7.

### PR DESCRIPTION
**Pull request by klaussilveira**
I'm planning to merge this PR on December 3rd, the latest official date for the PHP 7 GA release.

### CODE REVIEW

File: src/Adapter/AdapterInterface.php [hunk 1]
```diff
@@ -1,85 +1,23 @@
 <?php
 
+declare (strict_types = 1);
```
Comment (nesl247) [inline src/Adapter/AdapterInterface.php]: No blank line before this per PSR-12. While it is in draft, we should still adhere to it. https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md

Also, remove all the spaces in this. `declare(strict_types=1)`.

File: src/Adapter/AdapterInterface.php [hunk 2]
```diff
@@ -1,85 +1,23 @@
...
-
-    /**
-     * @param string $query
-     * @param array  $params
-     *
-     * @return int
-     */
-    public function execute($query, array $params = []);
-
+    public function __construct(string $driver, array $options, string $role);
```
Comment (nesl247) [inline src/Adapter/AdapterInterface.php:10]: While we don't necessarily need docblocks anymore for certain things, I'm not sure I agree with the removal of them. However, that aside, I think we should keep a new line between each method declaration.

File: src/Adapter/PdoAdapter.php [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare (strict_types = 1);
+
```
Comment (nesl247) [inline src/Adapter/PdoAdapter.php:3]: Same as previous message about declare.

File: src/Adapter/PdoAdapter.php [hunk 4]
```diff
@@ -7,43 +9,31 @@
 use Linio\Component\Database\Exception\DatabaseConnectionException;
 use Linio\Component\Database\Exception\DatabaseException;
 use Linio\Component\Database\Exception\InvalidQueryException;
+use PDO;
```
Comment (nesl247) [inline src/Adapter/PdoAdapter.php]: Per PSR-12, we also cannot import root level classes.

File: src/DatabaseAware.php [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare (strict_types = 1);
```
Comment (nesl247) [inline src/DatabaseAware.php]: Same as before.

File: src/DatabaseManager.php [hunk 6]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare (strict_types = 1);
```
Comment (nesl247) [inline src/DatabaseManager.php]: Once more.

File: src/DatabaseManager.php [hunk 7]
```diff
@@ -85,95 +74,49 @@ public function getConnections()
...
     }
 
-    /**
-     * @param string $query
-     * @param array  $params
-     *
-     * @return string
-     */
-    public function fetchValue($query, array $params = [])
+    public function fetchValue(string $query, array $params = [])
```
Comment (nesl247) [inline src/DatabaseManager.php:88]: This previous returned string but we are missing the return type hint here. Which was correct?

File: src/DatabaseManager.php [hunk 8]
```diff
@@ -215,12 +152,7 @@ public function rollBack()
         return true;
     }
 
-    /**
-     * @param string|null $name
-     *
-     * @return mixed
-     */
-    public function getLastInsertId($name = null)
+    public function getLastInsertId(string $name = null)
```
Comment (nesl247) [inline src/DatabaseManager.php:154]: Are we 100% sure this works? Null is definitely different than string. I haven't tested something like this yet in PHP 7.

File: src/Entity/Connection.php [hunk 9]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare (strict_types = 1);
```
Comment (nesl247) [inline src/Entity/Connection.php]: Once more...

File: src/Entity/LazyFetch.php [hunk 10]
```diff
@@ -1,28 +1,25 @@
 <?php
 
+declare (strict_types = 1);
```
Comment (nesl247) [inline src/Entity/LazyFetch.php]: Once more.

File: src/Entity/LazyFetch.php [hunk 11]
```diff
@@ -1,28 +1,25 @@
...
+
 namespace Linio\Component\Database\Entity;
 
 use Linio\Component\Database\Exception\DatabaseException;
+use PDOStatement;
```
Comment (nesl247) [inline src/Entity/LazyFetch.php]: Per PSR-12 we can't import this.

File: src/Entity/SlaveConnectionCollection.php [hunk 12]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare (strict_types = 1);
```
Comment (nesl247) [inline src/Entity/SlaveConnectionCollection.php]: Once more.

File: src/Adapter/PdoAdapter.php (hunk 4 above)
Comment (klaussilveira) [inline src/Adapter/PdoAdapter.php]: This implies changes to LSR as a whole, doesn't it?

File: src/DatabaseManager.php (hunk 8 above)
Comment (klaussilveira) [inline src/DatabaseManager.php:154]: Everything was tested on PHP 7 RC8.

File: src/DatabaseManager.php (hunk 7 above)
Comment (klaussilveira) [inline src/DatabaseManager.php:88]: null and bool can be returned.

File: src/Adapter/PdoAdapter.php (hunk 4 above)
Comment (nesl247) [inline src/Adapter/PdoAdapter.php]: LSR?

File: src/DatabaseManager.php (hunk 7 above)
Comment (nesl247) [inline src/DatabaseManager.php]: That's very odd, but ok. It really shouldn't be allowed at all at the language level.

File: src/DatabaseManager.php (hunk 8 above)
Comment (nesl247) [inline src/DatabaseManager.php:154]: I'm a bit surprised. It really shouldn't be allowed at all at the language level. But if they allow it, ok.

File: src/Adapter/PdoAdapter.php (hunk 4 above)
Comment (klaussilveira) [inline src/Adapter/PdoAdapter.php]: Linio Standards Resolution.

Comment (nesl247) [inline src/Adapter/PdoAdapter.php]: Yes, we will be making changes to the LSR as you have named it.

File: src/DatabaseManager.php (hunk 7 above)
Comment (nesl247) [inline src/DatabaseManager.php:88]: Ok cool. Just wanted to verify since the change being made broke the previous docblock.

**Pull request merged**